homeassistant/components/letpot/switch.py

```diff
@@ -0,0 +1,117 @@
+"""Support for LetPot switch entities."""
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from letpot.deviceclient import LetPotDeviceClient
+from letpot.models import DeviceFeature, LetPotDeviceStatus
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import LetPotConfigEntry
+from .coordinator import LetPotDeviceCoordinator
+from .entity import LetPotEntity
+
+# Each change pushes a 'full' device status with the change. The library will cache
+# pending changes to avoid overwriting, but try to avoid a lot of parallelism.
+PARALLEL_UPDATES = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class LetPotSwitchEntityDescription(SwitchEntityDescription):
+    """Describes a LetPot switch entity."""
+
+    value_fn: Callable[[LetPotDeviceStatus], bool | None]
+    set_value_fn: Callable[[LetPotDeviceClient, bool], Coroutine[Any, Any, None]]
+
+
+BASE_SWITCHES: tuple[LetPotSwitchEntityDescription, ...] = (
+    LetPotSwitchEntityDescription(
```

Comment:
I considered not setting a name for this entity and marking it as the [main feature](https://developers.home-assistant.io/docs/core/entity#example-of-a-switch-entity-which-is-the-main-feature-of-a-device), but decided against it because you'll rarely use this switch (should remain on while growing something, controlling light/pump etc will use individual entities)